contracts/aggregator/common/main.sol

```diff
@@ -0,0 +1,141 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+import './helpers.sol';
+import '@openzeppelin/contracts/utils/Address.sol';
+
+/**
+ * @title Flashloan.
+ * @dev Common aggregator implementation for all chains.
+ */
+
+contract AdminModule is HelpersCommon {
+    using SafeERC20 for IERC20;
+
+    event updateOwnerLog(address indexed oldOwner, address indexed newOwner);
+
+    event updateWhitelistLog(address indexed account, bool indexed isWhitelisted_);
+
+    event LogCollectRevenue(address to, address[] tokens, uint256[] amounts);
+
+    /**
+     * @dev owner gaurd.
+     * @notice owner gaurd.
+     */
+    modifier onlyOwner() {
+        require(msg.sender == owner, 'not-owner');
+        _;
+    }
+
+    /**
+     * @dev Update owner.
+     * @notice Update owner.
+     * @param newOwner_ address of new owner.
+     */
+    function updateOwner(address newOwner_) external onlyOwner {
+        address oldOwner_ = owner;
+        owner = newOwner_;
+        emit updateOwnerLog(oldOwner_, newOwner_);
+    }
+
+    /**
+     * @dev Function to add new routes.
+     * @notice Function to add new routes and implementations.
+     * @param _routes routes to add.
+     * @param _impls implementations of their respective routes.
+     */
+    function addNewRoutesAndEnable(uint256[] memory _routes, address[] memory _impls) public onlyOwner {
+        require(_routes.length == _impls.length, 'lengths-dont-match');
+        uint256 length = _routes.length;
+        for (uint256 i = 0; i < length; i++) {
+            require(routeToImpl[_routes[i]] == address(0), 'route-already-exists');
+            routeToImpl[_routes[i]] = _impls[i];
+            routeStatus[_routes[i]] = true;
+            routes.push(_routes[i]);
+        }
+    }
```

Comment:
Should we also add function to remove route from the `route` array?

Reply:
Added!